projects/igniteui-angular/src/lib/grids/grid/grid-keyBoardNav.spec.ts

```diff
@@ -1,465 +1,261 @@
+            const secondCell = GridFunctions.getRowCells(fix, 1)[0];
 
-        UIInteractions.triggerKeyDownWithBlur('arrowright', cell.nativeElement, true);
-        tick();
+            grid.onSelection.subscribe((event: IGridCellEventArgs) => {
+                selectedCell = event.cell;
+            });
+            firstCell.triggerEventHandler('focus', null);
+            tick(100);
+            fix.detectChanges();
 
-        fix.detectChanges();
-        expect(fix.componentInstance.selectedCell.value).toEqual('ALFKI');
-        expect(fix.componentInstance.selectedCell.column.field).toMatch('ID');
-        cell = cells[2];
+            expect(selectedCell.value).toEqual('Casey Houston');
+            expect(selectedCell.column.field).toMatch('Name');
 
-        UIInteractions.triggerKeyDownWithBlur('arrowleft', cell.nativeElement, true);
-        tick();
-        fix.detectChanges();
-        expect(fix.componentInstance.selectedCell.value).toEqual('Alfreds Futterkiste');
-        expect(fix.componentInstance.selectedCell.column.field).toMatch('CompanyName');
-        cell.triggerEventHandler('blur', {});
-        tick();
-        cell = cells[0];
+            UIInteractions.triggerEventHandlerKeyDownWithBlur('arrowdown', firstCell);
+            tick(100);
+            fix.detectChanges();
 
-        UIInteractions.triggerKeyDownWithBlur('arrowright', cell.nativeElement, true);
-        tick();
-        fix.detectChanges();
-        cell = cells[1];
+            expect(selectedCell.value).toEqual('Gilberto Todd');
+            expect(selectedCell.column.field).toMatch('Name');
 
-        UIInteractions.triggerKeyDownWithBlur('arrowright', cell.nativeElement, true);
-        tick();
-        fix.detectChanges();
-        expect(fix.componentInstance.selectedCell.value).toEqual('ALFKI');
-        expect(fix.componentInstance.selectedCell.column.field).toMatch('ID');
-    }));
+            UIInteractions.triggerEventHandlerKeyDownWithBlur('arrowup', secondCell);
+            tick(100);
+            fix.detectChanges();
 
-    it('should allow vertical keyboard navigation in pinned area.', fakeAsync(() => {
-        const fix = TestBed.createComponent(PinOnInitAndSelectionComponent);
-        fix.detectChanges();
-        const grid = fix.componentInstance.grid;
-        fix.detectChanges();
-        const cells = fix.debugElement.queryAll(By.css(CELL_CSS_CLASS));
-        let cell = cells[0];
+            expect(selectedCell.value).toEqual('Casey Houston');
+            expect(selectedCell.column.field).toMatch('Name');
+        }));
 
-        cell.nativeElement.dispatchEvent(new Event('focus'));
-        // cell.triggerEventHandler('focus', {});
+        it('should allow horizontal keyboard navigation between start pinned area and unpinned area.', fakeAsync(() => {
+            grid.getColumnByName('Name').pinned = true;
+            grid.getColumnByName('Company').pinned = true;
+            tick();
```

Comment:
Again it if there is not scrolling in this test the fakeAsync could be omitted completely.